salt-minion and salt-master displaying traceback when started

When starting/restarting salt-master/salt-minion on latest develop branch, it displays a traceback and gets hung. After a KeyBoard Interrupt, it displays ``OK``. 

```
root@nitin-salt-develop:~/salt$ service salt-master stop
Stopping salt-master daemon:                               [  OK  ]
root@nitin-salt-develop:~/salt$ service salt-master start
Starting salt-master daemon: Error in atexit._run_exitfuncs:
Traceback (most recent call last):
  File "/usr/lib64/python2.6/atexit.py", line 24, in _run_exitfuncs
    func(*targs, **kargs)
  File "/usr/lib64/python2.6/multiprocessing/util.py", line 269, in _exit_function
    p.join()
  File "/usr/lib64/python2.6/multiprocessing/process.py", line 117, in join
    assert self._parent_pid == os.getpid(), 'can only join a child process'
AssertionError: can only join a child process
Error in sys.exitfunc:
Traceback (most recent call last):
  File "/usr/lib64/python2.6/atexit.py", line 24, in _run_exitfuncs
    func(*targs, **kargs)
  File "/usr/lib64/python2.6/multiprocessing/util.py", line 269, in _exit_function
    p.join()
  File "/usr/lib64/python2.6/multiprocessing/process.py", line 117, in join
    assert self._parent_pid == os.getpid(), 'can only join a child process'
AssertionError: can only join a child process
^CError in atexit._run_exitfuncs:
Traceback (most recent call last):
  File "/usr/lib64/python2.6/atexit.py", line 24, in _run_exitfuncs
    func(*targs, **kargs)
  File "/usr/lib64/python2.6/multiprocessing/util.py", line 269, in _exit_function
    p.join()
  File "/usr/lib64/python2.6/multiprocessing/process.py", line 119, in join
    res = self._popen.wait(timeout)
  File "/usr/lib64/python2.6/multiprocessing/forking.py", line 122, in wait
    return self.poll(0)
  File "/usr/lib64/python2.6/multiprocessing/forking.py", line 107, in poll
    pid, sts = os.waitpid(self.pid, flag)
KeyboardInterrupt
Error in sys.exitfunc:
Error in sys.excepthook:
Traceback (most recent call last):
  File "/usr/lib/python2.6/site-packages/salt/log/setup.py", line 906, in __global_logging_exception_handler
    shutdown_multiprocessing_logging_listener()
  File "/usr/lib/python2.6/site-packages/salt/log/setup.py", line 800, in shutdown_multiprocessing_logging_listener
    __MP_LOGGING_QUEUE_PROCESS.terminate()
  File "/usr/lib64/python2.6/multiprocessing/process.py", line 111, in terminate
    self._popen.terminate()
  File "/usr/lib64/python2.6/multiprocessing/forking.py", line 139, in terminate
    os.kill(self.pid, signal.SIGTERM)
OSError: [Errno 3] No such process

Original exception was:
Traceback (most recent call last):
  File "/usr/lib64/python2.6/atexit.py", line 24, in _run_exitfuncs
    func(*targs, **kargs)
  File "/usr/lib64/python2.6/multiprocessing/util.py", line 269, in _exit_function
    p.join()
  File "/usr/lib64/python2.6/multiprocessing/process.py", line 119, in join
    res = self._popen.wait(timeout)
  File "/usr/lib64/python2.6/multiprocessing/forking.py", line 122, in wait
    return self.poll(0)
  File "/usr/lib64/python2.6/multiprocessing/forking.py", line 107, in poll
    pid, sts = os.waitpid(self.pid, flag)
KeyboardInterrupt
                                                           [  OK  ]
```

@cachedout Deployed a fresh new server and was able to replicate this:

```
root@cu-inf-d-slt01:~/salt$ service salt-master start
Starting salt-master daemon: /usr/lib64/python2.6/site-packages/Crypto/Util/number.py:57: PowmInsecureWarning: Not using mpz_powm_sec.  You should rebuild using libgmp >= 5 to avoid timing attack vulnerability.
  _warn("Not using mpz_powm_sec.  You should rebuild using libgmp >= 5 to avoid timing attack vulnerability.", PowmInsecureWarning)
Error in atexit._run_exitfuncs:
Traceback (most recent call last):
  File "/usr/lib64/python2.6/atexit.py", line 24, in _run_exitfuncs
    func(*targs, **kargs)
  File "/usr/lib64/python2.6/multiprocessing/util.py", line 269, in _exit_function
    p.join()
  File "/usr/lib64/python2.6/multiprocessing/process.py", line 117, in join
    assert self._parent_pid == os.getpid(), 'can only join a child process'
AssertionError: can only join a child process
Error in sys.exitfunc:
Traceback (most recent call last):
  File "/usr/lib64/python2.6/atexit.py", line 24, in _run_exitfuncs
    func(*targs, **kargs)
  File "/usr/lib64/python2.6/multiprocessing/util.py", line 269, in _exit_function
    p.join()
  File "/usr/lib64/python2.6/multiprocessing/process.py", line 117, in join
    assert self._parent_pid == os.getpid(), 'can only join a child process'
AssertionError: can only join a child process


^CError in atexit._run_exitfuncs:
Traceback (most recent call last):
  File "/usr/lib64/python2.6/atexit.py", line 24, in _run_exitfuncs
    func(*targs, **kargs)
  File "/usr/lib64/python2.6/multiprocessing/util.py", line 269, in _exit_function
    p.join()
  File "/usr/lib64/python2.6/multiprocessing/process.py", line 119, in join
    res = self._popen.wait(timeout)
  File "/usr/lib64/python2.6/multiprocessing/forking.py", line 122, in wait
    return self.poll(0)
  File "/usr/lib64/python2.6/multiprocessing/forking.py", line 107, in poll
    pid, sts = os.waitpid(self.pid, flag)
KeyboardInterrupt
Error in sys.exitfunc:
                                                           [  OK  ]
```

--- a/salt/utils/parsers.py
+++ b/salt/utils/parsers.py
@@ -558,7 +558,7 @@ class LogLevelMixIn(six.with_metaclass(MixInMeta, object)):
         # Setup extended logging right before the last step
         self._mixin_after_parsed_funcs.append(self.__setup_extended_logging)
         # Setup the multiprocessing log queue listener if enabled
-        self._mixin_after_parsed_funcs.append(self.__setup_mp_logging_listener)
+        self._mixin_after_parsed_funcs.append(self._setup_mp_logging_listener)
         # Setup the console as the last _mixin_after_parsed_func to run
         self._mixin_after_parsed_funcs.append(self.__setup_console_logger)
 
@@ -759,7 +759,7 @@ class LogLevelMixIn(six.with_metaclass(MixInMeta, object)):
     def _get_mp_logging_listener_queue(self):
         return log.get_multiprocessing_logging_queue()
 
-    def __setup_mp_logging_listener(self, *args):
+    def _setup_mp_logging_listener(self, *args):
         if self._setup_mp_logging_listener_:
             log.setup_multiprocessing_logging_listener(
                 self._get_mp_logging_listener_queue()
@@ -840,9 +840,17 @@ class DaemonMixIn(six.with_metaclass(MixInMeta, object)):
 
     def daemonize_if_required(self):
         if self.options.daemon:
+            if self._setup_mp_logging_listener_ is True:
+                # Stop the logging queue listener for the current process
+                # We'll restart it once forked
+                log.shutdown_multiprocessing_logging_listener()
+
             # Late import so logging works correctly
             import salt.utils
             salt.utils.daemonize()
+
+        # Setup the multiprocessing log queue listener if enabled
+        self._setup_mp_logging_listener()
 
     def is_daemonized(self, pid):
         import salt.utils.process

--- a/tests/integration/loader/loader.py
+++ b/tests/integration/loader/loader.py
@@ -451,7 +451,7 @@ class LazyLoaderModulePackageTest(TestCase):
     def setUp(self):
         self.opts = _config = minion_config(None)
         self.opts['grains'] = grains(self.opts)
-        self.tmp_dir = tempfile.mkdtemp(dir=tests.integration.TMP)
+        self.tmp_dir = tempfile.mkdtemp(dir=integration.TMP)
 
         dirs = _module_dirs(self.opts, 'modules', 'module')
         dirs.append(self.tmp_dir)

